flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ArchivedExecutionVertex.java

```diff
@@ -79,6 +96,11 @@ public ArchivedExecution getCurrentExecutionAttempt() {
         return currentExecution;
     }
 
+    @Override
+    public Collection<AccessExecution> getCurrentExecutions() {
+        return Collections.unmodifiableCollection(currentExecutions);
```

Comment:
I'm not too concerned about this because it's relatively lightweight and has been widely adopted in runtime code, as long as it will not be invoked too frequently (e.g. O(N) per RPC).